packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -101,18 +104,47 @@ export const PaymentUpdateForm = ({
     />
   );
 
-  const errorAlertBarContent = (
+  const billingAgreementErrorAlertBarContent = () => (
     <Localized
-      id="sub-route-invalid-payment-alert"
+      id="sub-route-missing-billing-agreement-payment-alert"
       elems={{ div: actionButton }}
     >
       <span>
-        Invalid payment information, there is an error with your account.{' '}
+        Invalid payment information; there is an error with your account.{' '}
         {actionButton}
       </span>
     </Localized>
   );
 
+  const fundingSourceErrorAlertBarContent = () => (
+    <Localized
+      id="sub-route-funding-source-payment-alert"
+      elems={{ div: actionButton }}
+    >
+      <span>
+        Invalid payment information; there is an error with your account. This
+        alert may take some time to clear after you successfully update your
+        information. {actionButton}
+      </span>
+    </Localized>
+  );
+
+  const getPaypalErrorAlertBarContent = () => {
+    switch (paypal_payment_error) {
+      case 'missing_agreement': {
```

Comment:
Ok, a suggestion.  I don't think it should block the merge, and I know you didn't originally introduce the type for `paypal_payment_error` for the frontend.

1. There's actually a type for it: https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts#L47  So we could move that to fxa-shared/subscriptions and use it as the type on the frontend as well.

2. Going further, we could additionally declare these values once somewhere and not use string literals for the cases here.

Reply:
I can do 1) easily. Do you have a suggestion for where to put these values for 2)? I could put it in `fxa-shared/subscriptions/metadata.ts`... Is the idea something like:
```ts
export const PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT: PaypalPaymentError = 'missing_agreement';
export const PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE: PaypalPaymentError = 'funding_source';
```
... and then import these consts where we use string literals?

Reply:
For 2) we could create a `fxa-shared/subscriptions/constants.ts`, or, stick them in the same `types.ts` since I was thinking of defining the type with them.


```
export const PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT = 'missing_agreement';
export const PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE = 'funding_source';

export type PaypalPaymentError = typeof PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT | typeof PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE;
```

Or, you can use an object:

```
export const PaypalPaymentErrors = {
  missingAgreement: 'missing_agreement',
  fundingSoruce: 'funding_source',
} as const;
export type PaypalPaymentError = typeof PaypalPaymentErrors[keyof typeof PaypalPaymentErrors];
```

I'll leave the style choice to you.